### PR TITLE
use deb gpg key from our server, not the gpg network

### PR DESCRIPTION
**Pull request by evgeni**
also ensure the key is refreshed if expired

### CODE REVIEW

**Comment by evgeni** [comment]
I do wonder if we should just use `apt::key` directly once, instead of twice in the definition of the repos.

File: manifests/repos/apt.pp [hunk 1]
```diff
@@ -10,7 +10,11 @@
   ::apt::source { $name:
     repos    => $repo,
     location => $location,
-    key      => $key,
+    key      => {
+      id     => $key,
+      source => 'http://deb.theforeman.org/foreman.asc',
+      ensure => 'refreshed',
```
Comment (ekohl) [inline manifests/repos/apt.pp:16]: Does this mean it fetches the key on every run?

File: manifests/repos/apt.pp [hunk 2]
```diff
@@ -10,7 +10,11 @@
   ::apt::source { $name:
     repos    => $repo,
     location => $location,
-    key      => $key,
+    key      => {
+      id     => $key,
+      source => 'http://deb.theforeman.org/foreman.asc',
```
Comment (ekohl) [inline manifests/repos/apt.pp:15]: Could you make the key also a parameter? Also wondering if we can fetch it over https so there is at least integrity.

**Comment by ekohl** [comment]
> I do wonder if we should just use `apt::key` directly once, instead of twice in the definition of the repos.

That probably makes sense.

File: manifests/repos/apt.pp (hunk 1 above)
Comment (evgeni) [reply manifests/repos/apt.pp:16]: according to https://forge.puppet.com/modules/puppetlabs/apt/reference#ensure-1:

Using 'refreshed' will make keys auto update when they have expired (assuming a new key exists on the key server).

File: manifests/repos/apt.pp [hunk 3]
```diff
@@ -4,25 +4,32 @@
   Variant[Enum['nightly'], Pattern['^\d+\.\d+$']] $repo,
   Variant[String, Hash] $key = 'AE0AF310E2EA96B6B6F4BD726F8600B9563278F6',
   Stdlib::HTTPUrl $location = 'https://deb.theforeman.org/',
+  Stdlib::HTTPUrl $key_location = 'https://deb.theforeman.org/foreman.asc',
 ) {
   include apt
 
+  ::apt::key { $name:
```
Comment (ekohl) [inline manifests/repos/apt.pp:11]: We didn't fix this in linting, but `::` is no longer needed
```suggestion
  apt::key { $name:
```

File: manifests/repos/apt.pp [hunk 4]
```diff
@@ -4,25 +4,32 @@
   Variant[Enum['nightly'], Pattern['^\d+\.\d+$']] $repo,
   Variant[String, Hash] $key = 'AE0AF310E2EA96B6B6F4BD726F8600B9563278F6',
```
Comment (ekohl) [inline manifests/repos/apt.pp:5]: Due to the new structure, I think this is now just a `String`:
```suggestion
  String $key = 'AE0AF310E2EA96B6B6F4BD726F8600B9563278F6',
```

File: manifests/repos/apt.pp [hunk 5]
```diff
@@ -4,25 +4,32 @@
...
   Stdlib::HTTPUrl $location = 'https://deb.theforeman.org/',
+  Stdlib::HTTPUrl $key_location = 'https://deb.theforeman.org/foreman.asc',
```
Comment (ekohl) [inline manifests/repos/apt.pp:7]: Mind keeping this with `$key` in the argument sorting?

**Review by ekohl — COMMENTED** [review]
I merged the systemd pin PR so if you update, please rebase.

**Comment by ehelms** [comment]
Pin camptocamp has been merged, so now you can drop it :)

**Review by ekohl — APPROVED** [review]

**Comment by ekohl** [comment]
Technically this changes a parameter but since it's a private API I don't consider it breaking, hence an enhancement.

**Pull request merged**